controllers/config/clientconfig_controller.go

```diff
@@ -205,3 +159,61 @@ func calculateHashes(ctx context.Context, anyClient client.Client, clientCfg con
 
 	return cfgHash, secretHash, nil
 }
+
+// initAdditionalCLusterConfig fetches the clientConfigs for additional clusters
+func initAdditionalCLusterConfig(r *ClientConfigReconciler, ctx context.Context, cCfg configapi.ClientConfig, mgr ctrl.Manager, watchNamespace string) (cluster.Cluster, error) {
+	uncachedClient := r.ClientCache.GetLocalNonCacheClient()
+	namespaces := strings.Split(watchNamespace, ",")
```

Comment:
You can pass `namespaces` as a parameter instead of `watchNamespace`, then the two lines that used the latter can be rewritten:
- `if strings.Contains(watchNamespace, ",") {` => `if len(namespaces) > 1 {`
- `o.Namespace = watchNamespace` => `o.Namespace = namespaces[0]`

I feel like we should probably check `len(namespaces) == 0` somewhere (the original code ignores it, but this probably causes a failure down the line).

Reply:
`namespaces` is created from a string using `strings.Split`, so the only way it's length would be `0` is if the namespace string passed in was empty. 

> I feel like we should probably check `len(namespaces) == 0` somewhere (the original code ignores it, but this probably causes a failure down the line).

I agree. I looked a little at the flow, but it seems a little tricky how it handles watchNamepsace. In [main.go](https://github.com/k8ssandra/k8ssandra-operator/blob/main/main.go#L109-L115) if there is an error getting `watchNamespace`, the error indicates it will watch all namespaces. This seems to be done by returning an [empty string](https://github.com/k8ssandra/k8ssandra-operator/blob/main/main.go#L289-L300) as the `watchNamespace`. So I guess a length of 0 does come into InitClientConfigs.....

Reply:
I guess my point is, it's a little more involved to handle length==0 it in this scope....

Reply:
> it will watch all namespaces. This seems to be done by returning an [empty string](https://github.com/k8ssandra/k8ssandra-operator/blob/main/main.go#L289-L300) as the watchNamespace

Ok, so in that case I think we need to change the first check to `if len(namespaces) != 1` to achieve the same result. Otherwise we'll enter the second branch and fail with an array index out of bounds.

Reply:
Actually forget this last comment.

`strings.split` returns the string itself if the separator is not present.
> If s does not contain sep and sep is not empty, Split returns a slice of length 1 whose only element is s.

So in this case we'll get an array of size 1 containing the empty string, and things work as expected.